compiler/rustc_query_system/src/dep_graph/graph.rs

```diff
@@ -66,13 +67,23 @@ pub struct MarkFrame<'a> {
     parent: Option<&'a MarkFrame<'a>>,
 }
 
-#[derive(Debug)]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
 pub(super) enum DepNodeColor {
     Green(DepNodeIndex),
-    Red,
+    Red(() /* DepNodeIndex */),
     Unknown,
 }
 
+impl DepNodeColor {
+    pub(super) fn as_green(self) -> Option<DepNodeIndex> {
```

Comment:
```suggestion
    pub(super) fn expect_green(self) -> DepNodeIndex {
```
The result is always unwrapped.